internal/channel/relay.go

```diff
@@ -0,0 +1,44 @@
+package channel
+
+import (
+	"sync"
+
+	"github.com/influxdata/telegraf"
+)
+
+type Relay struct {
+	sync.Mutex
+	src <-chan telegraf.Metric
+	dst chan<- telegraf.Metric
+}
+
+func NewRelay(src <-chan telegraf.Metric, dst chan<- telegraf.Metric) *Relay {
+	return &Relay{
+		src: src,
+		dst: dst,
+	}
+}
+
+func (r *Relay) Start() {
+	go func() {
+		for m := range r.src {
+			r.GetDest() <- m
+		}
+		close(r.GetDest())
```

Comment:
Possible race condition here, but after discussion only when telegraf is shutting down and to resolve this make sure telegraf can't accept `SetDest` calls when shutting down. Add clarifying comment regarding this behavior.

Reply:
fixed in 9794b45c